README.md

```diff
@@ -0,0 +1,182 @@
+
+### Manually, using `rouge`
+
+Image file can be created with `rouge` tool. This is a companion
+application for `moulin`.
+
+Right now it works only in standalone mode, so manual invocation is
+required.
+
+This XT product provides only one image: `full`.
+
+You can prepare the image by running
+
+```
+# rouge prod-devel-rcar-s4.yaml -i full
+```
+
+This will create file `full.img` in your current directory.
+
+Also you can write image directly to an SD card by running
+
+```
+# sudo rouge prod-devel-rcar.yaml -i full -so /dev/sdX
+```
+
+**BE SURE TO PROVIDE CORRECT DEVICE NAME**. `rouge` has no
+interactive prompts and will overwrite your device right away. **ALL
+DATA WILL BE LOST**.
+
+It is also possible to flash the image to the internal eMMC.
+For that you may want booting the board via TFTP and sharing DomD's
+root file system via NFS. Once booted it is possible to flash the image:
+
+```
+# dd if=/full.img of=/dev/mmcblk0 bs=1M
+```
+
+For more information about `rouge` check its
+[manual](https://moulin.readthedocs.io/en/latest/rouge.html).
+
+## U-boot environment
+
+It is possible to run the build from TFTP+NFS or eMMC. With NFS boot
+it is possible to configure board IP, server IP and NFS path for DomD.
+Please set the following environment for that:
+
+```
+setenv ipaddr '192.168.1.1'
+setenv serverip '192.168.1.100'
+setenv tftp_kernel_rswitch_load 'tftp 0x7a000000 Image'
+setenv tftp_dtb_rswitch_load 'tftp 0x48000000 r8a779f0-spider.dtb'
+setenv bootargs_rswitch 'root=/dev/nfs nfsroot=192.168.1.100:/srv/spider-rswitch,vers=3 ip=192.168.1.1:192.168.1.100::255.255.255.0:board:tsn0::::'
+setenv bootcmd_rswitch 'setenv bootargs $bootargs_rswitch; run tftp_dtb_rswitch_load; run tftp_kernel_rswitch_load; booti 0x7a000000 - 0x48000000'
+setenv bootcmd='run bootcmd_rswitch'
+```
+
+## Testing setup
+
+The setup can be tested using 2 PC or on the same PC. In case of the same PC, the interface connected to TSN2 should be moved to custom network namespace.
+
+## On board:
+
+```
+ifconfig tsn2 up 192.168.3.1
+ip route add 192.168.5.0/24 via 192.168.3.100
+echo 1 > /proc/sys/net/ipv4/ip_forward
+```
+
+## On host:
+
+### The same PC
+
+```
+sudo ip route add 192.168.5.0/24 via 192.168.1.1
+sudo ip netns add linuxhint
+# Interface, connected to tsn2
+sudo ifconfig enp1s0 down
+sudo ip link set enp1s0 netns linuxhint
+sudo ip netns exec linuxhint ifconfig enp1s0 up
+sudo ip netns exec linuxhint ifconfig enp1s0 192.168.3.100
+sudo ip netns exec linuxhint ifconfig virt0 hw ether C8:D7:4A:4E:47:50
+sudo ip netns exec linuxhint ip link add virt0 type dummy
+sudo ip netns exec linuxhint ifconfig virt0 hw ether C8:D7:4A:4E:47:50
+sudo ip netns exec linuxhint ifconfig virt0 up 192.168.5.100
+```
+
+### 2 PC
+
+On main host, connected to TSN0:
+```
+sudo ip route add 192.168.5.0/24 via 192.168.1.1
+```
+
+On host connected to TSN2:
+```
+sudo ifconfig enp1s0 192.168.3.100
+sudo ifconfig virt0 hw ether C8:D7:4A:4E:47:50
```

Comment:
Why do you need to hard-code MAC address?

Reply:
It was just an example, you can use any other valid MAC address. As the test setup was simplified, it was removed too.